compiler/rustc_typeck/src/check/fn_ctxt/suggestions.rs

```diff
@@ -287,6 +287,32 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         }
     }
 
+    /// When encountering a closure that captures variables, where a FnPtr is expected,
+    /// suggest a non-capturing closure
+    pub(in super::super) fn suggest_no_capture_closure(
+        &self,
+        err: &mut DiagnosticBuilder<'_>,
+        expected: Ty<'tcx>,
+        found: Ty<'tcx>,
+    ) {
+        if matches!(expected.kind(), ty::FnPtr(_)) {
+            if let ty::Closure(def_id, _) = found.kind() {
```

Comment:
This can instead be as follows to remove one level of indentation:

```suggestion
        if let (ty::FnPtr(_), ty::Closure(def_id, _)) = (expected.kind(), found.kind()) {
```